akka-actor-typed-tests/src/test/java/jdocs/akka/typed/OnMessageIntroTest.java

```diff
@@ -0,0 +1,284 @@
+      public final String message;
+
+      public PublishSessionMessage(String screenName, String message) {
+        this.screenName = screenName;
+        this.message = message;
+      }
+    }
+    // #chatroom-protocol
+
+    static interface SessionEvent {}
+
+    public static final class SessionGranted implements SessionEvent {
+      public final ActorRef<PostMessage> handle;
+
+      public SessionGranted(ActorRef<PostMessage> handle) {
+        this.handle = handle;
+      }
+    }
+
+    public static final class SessionDenied implements SessionEvent {
+      public final String reason;
+
+      public SessionDenied(String reason) {
+        this.reason = reason;
+      }
+    }
+
+    public static final class MessagePosted implements SessionEvent {
+      public final String screenName;
+      public final String message;
+
+      public MessagePosted(String screenName, String message) {
+        this.screenName = screenName;
+        this.message = message;
+      }
+    }
+
+    static interface SessionCommand {}
+
+    public static final class PostMessage implements SessionCommand {
+      public final String message;
+
+      public PostMessage(String message) {
+        this.message = message;
+      }
+    }
+
+    private static final class NotifyClient implements SessionCommand {
+      final MessagePosted message;
+
+      NotifyClient(MessagePosted message) {
+        this.message = message;
+      }
+    }
+    // #chatroom-protocol
+    // #chatroom-behavior
+
+    public static Behavior<RoomCommand> create() {
+      return Behaviors.setup(ChatRoomBehavior::new);
+    }
+
+    public static class ChatRoomBehavior extends AbstractOnMessageBehavior<RoomCommand> {
+      final List<ActorRef<SessionCommand>> sessions = new ArrayList<>();
+
+      private ChatRoomBehavior(ActorContext<RoomCommand> context) {
+        super(context);
+      }
+
+      @Override
+      public Behavior<RoomCommand> onMessage(RoomCommand msg) throws UnsupportedEncodingException {
+        /* From Java 16 onward, various features broadly described as "pattern matching"
+         * may prove useful here in lieu of the explicit instanceof checks and casts:
+         *
+         * Java 16 onward: JEP 394 (https://openjdk.java.net/jeps/394) =>
+         *   if (msg instanceof GetSession gs) {
+         *     return onGetSession(gs);
+         *   } else if (msg instanceof PublishSessionMessage psm) {
+         *     return onPublishSessionMessage(psm);
+         *   }
+         *
+         * Java 17 onward: JEP 406 (https://openjdk.org/jeps/406) =>
+         *   switch(msg) {
+         *     // NB: default clause omitted for brevity - JEP 409 (https://openjdk.org/jeps/409)
+         *     // may allow not including a default clause
+         *     case GetSession gs:
+         *       return onGetSession(gs);
+         *
+         *     case PublishSessionMessage psm:
+         *       return onPublishSessionMessage(psm);
+         *   }
```

Comment:
I think it would be better to show this in the documentation, even though it's not compiled. Should be possible with some paradox tagging:
``` 
/*
// #chatroom-behavior
            switch(msg) {
               // NB: default clause omitted for brevity - JEP 409 (https://openjdk.org/jeps/409)
               // may allow not including a default clause
               case GetSession gs:
                 return onGetSession(gs);
          
               case PublishSessionMessage psm:
                 return onPublishSessionMessage(psm);
             }
// #chatroom-behavior
*/
```

Reply:
and same for the other snippets